### DISCUSSION TITLE
Using SolidJS with TypeScript - Normal functions not assignable to Component type

### DISCUSSION THREAD

**Discussion by 174n** [solidjs/solid#1645] [Q&A]
Hello everyone,

I recently started using SolidJS with TypeScript and encountered an issue when trying to define components with normal functions. In JavaScript, I was able to define components using normal functions, but when switching to TypeScript, I found that I had to use arrow functions instead.

If I define a component using a normal function, I receive the following error: "Type 'Element' is not assignable to type 'Component<{}>'". It seems that TypeScript is expecting the component to be an arrow function instead.

I'm wondering if anyone knows why this is the case and if there is a way to use normal functions instead of arrow functions. I'd appreciate any insight or advice!

Thank you in advance.

**Comment by lxsmnsyc** [comment]
Without an example code/repro, there's no way to answer your concern.

**Reply by quazar-omega** [reply]
With this example:
```tsx
import { Component } from 'solid-js'

function Example(): Component {

	return <div>Test</div>
}

export default Example
```

I get this:
```
Type 'Element' is not assignable to type 'Component'.
  Type 'number' is not assignable to type 'Component'.
```

<details>
<summary>
Full from VSCode
</summary>

```json
[{
	"resource": "Example.tsx",
	"owner": "typescript",
	"code": "2322",
	"severity": 8,
	"message": "Type 'Element' is not assignable to type 'Component'.\n  Type 'number' is not assignable to type 'Component'.",
	"source": "ts",
	"startLineNumber": 6,
	"startColumn": 2,
	"endLineNumber": 6,
	"endColumn": 24
}]
```

**Reply by atk** [reply]
Component is a function that returns a JSX.Element, so you probably want:
```ts
import { JSX } from 'solid-js'

function Example(): JSX.Element {
	return <div>Test</div>
}

export default Example
```

**Reply by quazar-omega** [reply]
Yes, I had tried that and it does work, but then why does the mismatch between the arrow function type annotation and the normal function annotation happen?

Edit:
I just realized, the corresponding arrow function would be this actually
```tsx
import { Component, JSX } from 'solid-js'

const Example: Component = (): JSX.Element => {
	return <div>Test</div>
}

export default Example
```

**Reply by atk** [reply]
In the latter example, you can leave out the return type, as it is already set in Component.

**Reply by quazar-omega** [reply]
Yep, you're right!  
Just found it helpful to wrap my head around it, thanks for the heads up
